src/QualificationRunner.Core/Services/QualificationRunner.cs

```diff
@@ -90,21 +136,28 @@ private Task<SaticFiles> copyStaticFiles(dynamic qualificationPlan)
 
          //Observed Data
          IReadOnlyList<ObservedDataMapping> observedDataSets = getStaticObservedDataSetFrom(qualificationPlan);
-         staticFiles.ObservedDatSets = observedDataSets.Select(copyObservedData).ToArray();
+         staticFiles.ObservedDatSets = await Task.WhenAll(observedDataSets.Select(copyObservedData));
 
-         return Task.FromResult(staticFiles);
+         return staticFiles;
       }
 
       private IReadOnlyList<ObservedDataMapping> getStaticObservedDataSetFrom(dynamic qualificationPlan) => GetListFrom<ObservedDataMapping>(qualificationPlan.ObservedDataSets);
 
-      private string errorMessageFrom(IEnumerable<QualificationRunResult> invalidResults) => invalidResults.Select(x => Errors.ProjectConfigurationNotValid(x.Project, x.LogFile)).ToString("\n");
+      private string errorMessageFrom(IEnumerable<QualificationRunResult> invalidResults) => invalidResults.Select(x => ProjectConfigurationNotValid(x.Project, x.LogFile)).ToString("\n");
 
-      private ObservedDataMapping copyObservedData(ObservedDataMapping observedDataMapping)
+      private async Task<ObservedDataMapping> copyObservedData(ObservedDataMapping observedDataMapping)
       {
-         var observedDataFilePath = absolutePathFrom(_runOptions.ConfigurationFolder, observedDataMapping.Path);
-         var fileInfo = new FileInfo(observedDataFilePath);
-         if (!fileInfo.Exists)
-            throw new QualificationRunException(ObservedDataFileNotFound(observedDataFilePath));
+         Task<string> downloadRemoteObservedData() => downloadRemoteFile(observedDataMapping.Path, $"{observedDataMapping.Id}{Filter.CSV_EXTENSION}", OBSERVED_DATA_DOWNLOAD_FOLDER, observedDataMapping.Id, "Observed Data");

```

Comment:
Same here: please use original file name fow download and not observedDataMapping.Id